src/odemis/gui/cont/streams.py

```diff
@@ -544,6 +548,28 @@ def _on_metadata_btn(self, evt):
                         v = u"[%s … %s]" % (u", ".join(str(a) for a in v[:20]), u", ".join(str(a) for a in v[-20:]))
                     text += u"%s: %s\n" % (key, v)
 
+        # only display extra settings once
+        if model.MD_EXTRA_SETTINGS in raw[0].metadata:
+            text += u"\n======================================\nHardware Settings\n"
+            text += u"======================================\n"
+            for comp, vas in md[model.MD_EXTRA_SETTINGS].items():
+                if vas != {}:
+                    text += u"Component %s:\n" % comp
+                for name, value in vas.items():
+                    data = value[0]
+                    unit = value[1] or ""  # don't display 'None'
+                    unit = unit if data else ""  # don't display unit if data is None (None Hz doesn't make sense)
+                    if isinstance(data, dict):
+                        if value != {}:
```

Comment:
Probably `value` -> `data` ? (not if you rename the variable ;-) )
Also you should be able to write `if data:`.